lib/config/guiconfiguration.go

```diff
@@ -113,6 +115,22 @@ func (c GUIConfiguration) URL() string {
 	return u.String()
 }
 
+// SetHashedPassword hashes the given plaintext password and stores the new hash.
+func (c *GUIConfiguration) SetHashedPassword(password string) error {
```

Comment:
SetHashed implies to accept an already hashed password.

HashAndSetPassword would be better?